client/src/search/Typeahead/Typeahead.js

```diff
@@ -332,74 +322,26 @@ export class Typeahead extends React.Component {
   }
 
   get derived_menu_state() {
-    const { page_size } = this.props;
-    const { pagination_cursor, matching_results_by_page } = this.state;
-
-    const total_matching_results = _.flatten(matching_results_by_page).length;
-
-    const results_on_page = matching_results_by_page[pagination_cursor] || [];
+    const { matching_results } = this.state;
 
-    const page_range_start = pagination_cursor * page_size + 1;
-    const page_range_end = page_range_start + results_on_page.length - 1;
+    const total_matching_results = _.flatten(matching_results).length;
 
-    const remaining_results =
-      (pagination_cursor + 1) * page_size < total_matching_results
-        ? total_matching_results - (pagination_cursor + 1) * page_size
-        : 0;
-
-    const next_page_size =
-      remaining_results < page_size ? remaining_results : page_size;
-
-    const needs_pagination_up_control = pagination_cursor > 0;
-    const needs_pagination_down_control =
-      page_range_end < total_matching_results;
-
-    const total_menu_items = (() => {
-      if (needs_pagination_up_control && needs_pagination_down_control) {
-        return results_on_page.length + 2;
-      } else if (needs_pagination_up_control || needs_pagination_down_control) {
-        return results_on_page.length + 1;
-      } else {
-        return results_on_page.length;
-      }
-    })();
+    const total_menu_items = matching_results.length;
```

Comment:
There aren't any non-result menu items, would think `total_menu_items === total_matching_results`. Actually, the flatten was back from when `matching_results` was `matching_results_by_page`, so I'm pretty sure these ARE equal.